### PR TITLE
fix(geometry): deterministic fallback in ParametrizedLine.intersect for parallel case (#3870)

### LINKED ISSUE #3870: [Bug]: Remove redundant empty tensor initializations in kornia.geometry.conversions
### 🐛 Describe the bug

In `kornia.geometry.conversions`, several functions (`quaternion_to_axis_angle`, `quaternion_log_to_exp`, `quaternion_exp_to_log`) contain redundant initializations like `var_name: torch.Tensor = torch.tensor([])` right before being reassigned.

For example:
```python
    q1: torch.Tensor = torch.tensor([])
    q2: torch.Tensor = torch.tensor([])
    q3: torch.Tensor = torch.tensor([])
    cos_theta: torch.Tensor = torch.tensor([])

    cos_theta = quaternion[..., 0]
    q1 = quaternion[..., 1]
    q2 = quaternion[..., 2]
    q3 = quaternion[..., 3]
```

This causes unnecessary CPU memory allocations for temporary empty tensors that are immediately discarded, leading to unnecessary garbage collection and memory overhead.

### 🔄 Steps to Reproduce

```bash
1. Open `kornia/geometry/conversions.py`.
2. Inspect functions `quaternion_to_axis_angle`, `quaternion_log_to_exp`, `quaternion_exp_to_log`.
3. Observe redundant initializations like `q1: torch.Tensor = torch.tensor([])`.
```

### 💻 Minimal Code Example

```python
q1: torch.Tensor = torch.tensor([]) # Redundant CPU memory allocation
    q2: torch.Tensor = torch.tensor([]) # Redundant CPU memory allocation
    q3: torch.Tensor = torch.tensor([]) # Redundant CPU memory allocation
    cos_theta: torch.Tensor = torch.tensor([]) # Redundant CPU memory allocation

etc.
```

### ✅ Expected behavior

remove 'torch.Tensor([])`

```python
# Expected behavior
cos_theta = quaternion[..., 0]
q1 = quaternion[..., 1]
q2 = quaternion[..., 2]
q3 = quaternion[..., 3]
```

or only declare

```python
# Expected behavior
q1: torch.Tensor
q2: torch.Tensor
q3: torch.Tensor

cos_theta = quaternion[..., 0]
q1 = quaternion[..., 1]
q2 = quaternion[..., 2]
q3 = quaternion[..., 3]
```

### ❌ Actual behavior

No runtime error is thrown, but empty `torch.tensor([])` instances are allocated on CPU memory unnecessarily and immediately discarded right after assignment. 

This causes redundant memory allocation overhead and triggers unnecessary garbage collection during execution.

### 🔧 Environment

```shell
- Kornia version: main branch
- PyTorch version: 2.x
- Python version: 3.11
- OS (e.g., Linux, macOS, Windows): Window
- Installation method (pip, conda, from source): from source
- CUDA/cuDNN version (if applicable): N/A
- GPU model (if applicable): N/A
```

### 📝 Additional context

_No response_

### 🤝 Contribution Intent

- [x] I plan to submit a PR to fix this bug
- [ ] I'm reporting this bug but not planning to fix it

### PR DESCRIPTION
**Pull request by Prakhar00001**
Fixes #3870

Replaces redundant/uninitialized `torch.empty_like` with `torch.zeros_like` in `kornia/geometry/line.py` to prevent uninitialized memory allocation during line geometry calculations.

### CODE REVIEW

**Review by edgarriba — DISMISSED** [review]
Is this a bug ? Provide a test showing crash before and fixed after

**Comment by ducha-aiki** [comment]
Pushed a maintainer addition: `test_intersect_plane_parallel` pins the deterministic contract for the degenerate branch (line parallel to plane → `lambda=0`, point = line origin), which was previously uninitialized `empty_like` memory. Retitled to `fix(geometry)` since this changes behavior, not style. Local run: `38 passed` for tests/geometry/test_line.py.

**Comment by Prakhar00001** [comment]
Thanks @ducha-aiki for pinning the deterministic behavior test and refining the PR title! 
Appreciate the maintainer addition.

File: kornia/geometry/line.py [hunk 1]
```diff
@@ -172,7 +172,7 @@ def intersect(self, plane: Hyperplane, eps: float = 1e-6) -> Tuple[torch.Tensor,
         res_lambda = torch.where(
             dot_prod_mask,
             -(plane.offset + batched_dot_product(plane.normal.data, self.origin.data)) / dot_prod,
-            torch.empty_like(dot_prod),
+            torch.zeros_like(dot_prod),
         )
```
[the remark this answers is not in the document]
Comment (ducha-aiki) [reply kornia/geometry/line.py:180]: The degenerate-case return is now documented in the docstring (lambda 0, line origin) and pinned by test_intersect_plane_parallel. Returning a validity mask would change the public return signature — that is already tracked by the module's existing TODO ('possibly return a mask') and is out of scope for this minimal determinism fix.

**Review by ducha-aiki — APPROVED** [review]
